stdlib/queue.gr

```diff
@@ -295,4 +327,43 @@ provide module Immutable {
       { forwards, backwards } => List.length(forwards) + List.length(backwards),
     }
   }
+
+  /**
+   * Converts a queue into a list of its elements.
+   *
+   * @param queue: The queue to convert
+   * @returns A list containing all queue values
+   *
+   * @since v0.6.0
+   */
+
+  provide let toList = queue => {
+    let mut queue = queue
+    let mut lst = []
+    while (!isEmpty(queue)) {
+      match (peek(queue)) {
+        Some(value) => {
+          queue = pop(queue)
+          lst = [value, ...lst]
+        },
+        None => break,
```

Comment:
Doesn't a `None` on peek indicate an empty queue? Why even do the `isEmpty` check?

But also, since you have access to the internals, can't this be implemented something like `List.concat(queue.forwards, List.reverse(queue.backwards))`?